### PR TITLE
RS-306 - ETL for newtab interactions 

### PR DESCRIPTION
**Pull request by ANich**
ETL for initial newtab interactions table.

### CODE REVIEW

File: dags.yaml [hunk 1]
```diff
@@ -766,3 +766,22 @@ bqetl_fog_decision_support:
...
+    email:
+      - telemetry-alerts@mozilla.com
+      - anicholson@mozilla.com
+    email_on_failure: true
+    email_on_retry: true
+    end_date: null
+    owner: anicholson@mozilla.com
+    retries: 2
+    retry_delay: 30m
+    start_date: '2022-07-01'
```
Comment (kik-kik) [inline dags.yaml:782]:
```suggestion
    start_date: "2022-07-01"
```

^^ just for consistency

File: sql/moz-fx-data-shared-prod/telemetry_derived/newtab_interactions_v1/query.sql [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+    normalized_os,
+    normalized_os_version,
+    normalized_country_code,
+    normalized_channel,
+    client_info.app_display_version,
+    mozfun.map.get_key(event_details, "source") AS newtab_open_source,
+    metrics.string.search_engine_private_engine_id AS default_search_engine,
+    metrics.string.search_engine_default_engine_id AS default_private_search_engine,
+    ping_info.experiments,
+        -- ??? private_browsing_mode
```
Comment (kik-kik) [inline sql/moz-fx-data-shared-prod/telemetry_derived/newtab_interactions_v1/query.sql:74]: what are the `???`?

File: sql/moz-fx-data-shared-prod/telemetry_derived/newtab_interactions_v1/query.sql [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+    normalized_channel,
+    ping_info,
+    extra AS event_details,
+    metrics
+  FROM
+    `moz-fx-data-shared-prod.firefox_desktop_stable.newtab_v1`,
+    UNNEST(events)
+  WHERE
+    DATE(submission_timestamp) = @submission_date
+    AND category IN ('newtab', 'topsites', 'newtab.search', 'newtab.search.ad')
```
Comment (kik-kik) [inline sql/moz-fx-data-shared-prod/telemetry_derived/newtab_interactions_v1/query.sql:21]: for safety should we be doing `LOWER(category)`?
same for name below?

File: sql/moz-fx-data-shared-prod/telemetry_derived/newtab_interactions_v1/metadata.yaml [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+  time_partitioning:
+    field: submission_date
+    type: day
+    require_partition_filter: true
+    expiration_days: null
+  clustering:
+    fields:
+    - channel
+    - country
+references: {}
```
Comment (kik-kik) [inline sql/moz-fx-data-shared-prod/telemetry_derived/newtab_interactions_v1/metadata.yaml:23]: Should `moz-fx-data-shared-prod.firefox_desktop_stable.newtab_v1` be added to references? I'm not actually quite sure how that key works...

File: sql/moz-fx-data-shared-prod/telemetry_derived/newtab_interactions_v1/metadata.yaml [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+bigquery:
+  time_partitioning:
+    field: submission_date
+    type: day
+    require_partition_filter: true
+    expiration_days: null
+  clustering:
+    fields:
+    - channel
+    - country
```
Comment (kik-kik) [inline sql/moz-fx-data-shared-prod/telemetry_derived/newtab_interactions_v1/metadata.yaml:22]: Was this meant to be `country_code`? I tried to do search and could not find `country` in the SQL.

File: sql/moz-fx-data-shared-prod/telemetry_derived/newtab_interactions_v1/schema.yaml [hunk 6]
```diff
@@ -0,0 +1,103 @@
...
+      type: STRING
+    - fields:
+      - mode: NULLABLE
+        name: type
+        type: STRING
+      mode: NULLABLE
+      name: extra
+      type: RECORD
+    mode: NULLABLE
+    name: value
```
Comment (kik-kik) [inline sql/moz-fx-data-shared-prod/telemetry_derived/newtab_interactions_v1/schema.yaml:33]: The schama is a bit hard to read. Could we maybe do something as follows:

- name:
  mode: 
  type:
  fields:

so sort of reading from top to bottom?

**Review by kik-kik — APPROVED** [review]

File: dags.yaml (hunk 1 above)
Comment (ANich) [reply dags.yaml:782]: Seems the other `start_date`'s use single quotes. It's probably in the generation template

File: sql/moz-fx-data-shared-prod/telemetry_derived/newtab_interactions_v1/metadata.yaml (hunk 5 above)
Comment (ANich) [reply sql/moz-fx-data-shared-prod/telemetry_derived/newtab_interactions_v1/metadata.yaml:22]: Yes it was good catch!

File: sql/moz-fx-data-shared-prod/telemetry_derived/newtab_interactions_v1/query.sql (hunk 3 above)
Comment (ANich) [reply sql/moz-fx-data-shared-prod/telemetry_derived/newtab_interactions_v1/query.sql:21]: These are usually standardized/version-controlled in the client I think this should be fine. Though I will add a link to the Glean dictionary entry

File: sql/moz-fx-data-shared-prod/telemetry_derived/newtab_interactions_v1/query.sql (hunk 2 above)
Comment (ANich) [reply sql/moz-fx-data-shared-prod/telemetry_derived/newtab_interactions_v1/query.sql:74]: These are meant to be added in a future PR (upstream data changes are required)

**Pull request merged**